src/hackney_pool.erl

```diff
@@ -127,8 +127,7 @@ do_checkout(Requester, Host, _Port, Transport, #client{options=Opts,
     {error, Reason} ->
       {error, Reason};
     {'EXIT', {timeout, _}} ->
-      % socket will still checkout so to avoid deadlock we send in a cancellation
-      gen_server:cast(Pool, {checkout_cancel, Connection, RequestRef}),
+      %% checkout should be canceled by the caller via hackney_manager
```

Comment:
No need to cancel it here, as we now can rely on  eventually caneling it by hackney_manager.